text/0000-no-entry.md

```diff
@@ -0,0 +1,149 @@
+- Feature Name: `no_entry`
+- Start Date: 2019-07-23
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add a new top-level attribute `no_entry`, to omit the platform entry point
+symbol, allowing the user to write their own.
+
+# Motivation
+[motivation]: #motivation
+
+For some low-level systems-programming use cases, a user needs to have full
+control of a process from the very first instruction, by specifying the
+entry-point symbol. Examples include kernels or firmware where the entry-point
+symbol may represent initial boot code, applications running under an operating
+system but with unusual startup requirements, or programs creating or running
+in sandbox environments.
+
+The target toolchain will commonly supply an entry-point symbol (e.g.
+`_start`) by default. As a result, attempting to define an entry-point symbol
+typically results in a link error. Addressing this currently requires linking
+manually with an toolchain-specific argument like `-nostartfiles`.
+
+This proposal introduces a top-level attribute `no_entry`, which will cause the
+Rust toolchain to omit the entry-point symbol. Specifying `#![no_entry]` allows
+(and requires) the user to supply such a symbol.
+
+(Please note that this is entirely distinct from the Rust-supplied startup hook
+`#[start]`.)
```

Comment:
Can this section say something about why compiling code as a static library and linking afterwards is insufficient?

Reply:
@nagisa The later Rationale section discusses that:

> Today, people do this either by linking their binary manually, or by specifying non-portable linker arguments in `.cargo/config`. Introducing the `no_entry` attribute substantially simplifies creating such programs, and avoids target-specific build steps.